src/Orchard/Security/IAccountValidationService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orchard.Localization;
+
+namespace Orchard.Security {
+    public interface IAccountValidationService : IDependency {
```

Comment:
Why so many methods and not just one method with a `UserVallidationContext` ?
You could do validation that needs many of these properties.

Reply:
I had thought of that, but decided against it. 
The reason was to provide building blocks for a method like that so that with a single implementation of this service it would be possible to build different validation "combinations".
For example, the `ValidateRegistration` method of `AccountController` uses these methods, then adds a check for the password confirmation. 
Suppose in the same tenant I want to have a newsletter service, that would allowme to send newsletters to people that are not registered on the tenant. The "registration" for a newsletter may skip the password and username validations entirely, but may still check for unicity (I would add this in the method from the controller, like it's done in the `AccountController`). 
If I had a single big method with an implementation doing all the validations, that would be injected in both of those controllers, and then I would require the passed context to have a lot of properties to control the validation steps to perform, and possibly their order.

Reply:
I disagree. If you need a newsletter module to create accounts then you should handle it by either using the same default logic that is used for new accounts, or create a custom one.

Also by separating all calls you can't ensure a rule where a password for a user can't be reused. Just an example.